From 70ba7ab5f0ff60a5b8cf4172b831bfdab1d1a55f Mon Sep 17 00:00:00 2001
From: Philipp Kerling <pkerling@casix.org>
Date: Mon, 11 Jun 2018 01:33:55 +0200
Subject: [PATCH] Util::GetMatchingSource: Unset protocol options for matching

If the source URL has protocol options set, files in the source
will never match without removing them first.
---
 xbmc/Util.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/Util.cpp b/xbmc/Util.cpp
index 1d6915c1e32f..11951a8b13fa 100644
--- a/xbmc/Util.cpp
+++ b/xbmc/Util.cpp
@@ -1223,6 +1223,7 @@ int CUtil::GetMatchingSource(const std::string& strPath1, VECSOURCES& VECSOURCES
   // and ends with a trailing slash so as not to match a substring
   CURL urlDest(strPath);
   urlDest.SetOptions("");
+  urlDest.SetProtocolOptions("");
   std::string strDest = urlDest.GetWithoutUserDetails();
   ForceForwardSlashes(strDest);
   if (!URIUtils::HasSlashAtEnd(strDest))
@@ -1259,6 +1260,7 @@ int CUtil::GetMatchingSource(const std::string& strPath1, VECSOURCES& VECSOURCES
       // and ends with a trailing slash so as not to match a substring
       CURL urlShare(path);
       urlShare.SetOptions("");
+      urlShare.SetProtocolOptions("");
       std::string strShare = urlShare.GetWithoutUserDetails();
       ForceForwardSlashes(strShare);
       if (!URIUtils::HasSlashAtEnd(strShare))
